ui/js/filters.js

```diff
@@ -33,6 +33,45 @@ treeherder.filter('stripHtml', function() {
     };
 });
 
+treeherder.filter('linkifyURLs', function() {
+    return function(input) {
+        var str = input || '';
+        var urlpattern = /(\b(https?):\/\/[-A-Z0-9+&@#\/%?=~_|!:,.;]*[-A-Z0-9+&@#\/%=~_|])/gim;
+        str = str.replace(urlpattern, '<a href="$1" target="_blank">$1</a>');
+        return str;
+    };
+});
+
+treeherder.filter('linkifyClassifications', ['ThRepositoryModel', function(ThRepositoryModel) {
+    return function(input, projectName) {
+        var str = input || '';
+        str = str.trim();
+
+        var repo = ThRepositoryModel.getRepo(projectName);
+
+        if(repo.dvcs_type === "hg" && isAlphaNumeric(str) && (str.length === 12 || str.length === 40)) {
+            var hg_url = '<a href="' + repo.url + '/rev/' + str + '">' + str + '</a>';
+            str = hg_url;
+        }
+
+        return str;
+    };
+}]);
+
+function isAlphaNumeric(str) {
+    var code, i, len;
+
+    for (i = 0, len = str.length; i < len; i++) {
```

Comment:
Suppose this could also be a regex. Read somewhere online that this was more efficient than a regex that checks for alphanumeric. Not really set on this, happy to change it.